src/xenia/gpu/shader.cc

```diff
@@ -49,41 +49,58 @@ std::pair<std::string, std::string> Shader::Dump(const std::string& base_path,
     xe::filesystem::CreateFolder(target_path);
   }
 
-  char txt_file_name[kMaxPath];
-  std::snprintf(txt_file_name, xe::countof(txt_file_name),
-                "%s/shader_%s_%.16" PRIX64 ".%s", base_path.c_str(),
-                path_prefix, ucode_data_hash_,
-                shader_type_ == ShaderType::kVertex ? "vert" : "frag");
-  FILE* f = fopen(txt_file_name, "wb");
-  if (f) {
-    fwrite(translated_binary_.data(), 1, translated_binary_.size(), f);
-    fprintf(f, "\n\n");
-    auto ucode_disasm_ptr = ucode_disassembly().c_str();
-    while (*ucode_disasm_ptr) {
-      auto line_end = std::strchr(ucode_disasm_ptr, '\n');
-      fprintf(f, "// ");
-      fwrite(ucode_disasm_ptr, 1, line_end - ucode_disasm_ptr + 1, f);
-      ucode_disasm_ptr = line_end + 1;
-    }
-    fprintf(f, "\n\n");
-    if (!host_disassembly_.empty()) {
-      fprintf(f, "\n\n/*\n%s\n*/\n", host_disassembly_.c_str());
+  std::string txt_file_name;
+  int required_size =
+      std::snprintf(nullptr, 0, "%s/shader_%s_%.16" PRIX64 ".%s",
```

Comment:
The way you ended up changing it here is excessively verbose. I think it would be better to `std::snprintf` the data hash into a static buffer then concatenate the string as normal instead of this giant repetition of code.

Reply:
Can a reliable assumption be made about max length of this hash? If so, I can of course do it.

Reply:
`"%.16" PRIX64` is `%.16llx` -- so yes.

Reply:
Yes, but `path_prefix` doesn't seem to have size defined, so I would only save on reformatting `ucode_data_hash_` and maybe extension. Is it worth it?

Reply:
Nevermind, gave this a second look after a while and now I get what do you mean!